crates/holochain_types/src/app/app_bundle.rs

```diff
@@ -296,13 +296,27 @@ impl AppRoleResolution {
     /// Return the IDs of new cells to be created as part of the resolution.
     /// Does not return existing cells to be reused.
     pub fn cells_to_create(&self) -> Vec<(CellId, Option<MembraneProof>)> {
+        let provisioned = self
```

Comment:
This is the fix, just look at the roles and filter out any cells which aren't supposed to be created